Pcap++/header/PcapLiveDevice.h

```diff
@@ -174,6 +174,12 @@ namespace pcpp
 			 */
 			int packetBufferSize;
 
+            /**
+             * Set the direction for capture packets. You can read more here:
+             * https://www.tcpdump.org/manpages/pcap.3pcap.html.
+             */
+            pcap_direction_t direction;
```

Comment:
can you please replace `pcap_direction_t` with a structure defined in this file? wherever possible I'd like to avoid using libpcap structs in header files to make it easier for apps/libs to link with PcapPlusPlus

Reply:
So I should create a structure like
```
struct PcapThread
{
    pthread_t pthread;
};
```
So it can be
```
struct PcapDirection
{
    pcap_direction_t direction.
}
```
And could you please give me some advice on how to write a unit-tests for this? I just know I can test it with `tcpdump`, because I can compare the results got from `tcpdump` with it. But I don't  know how to do this in a test function.

Reply:
Yes I think defining a new enum like `PcapDirection` is a better approach. You can then define a method in `PcapLiveDevice.cpp` to convert between this enum and `pcap_direction_t`.

Regarding unit-tests - maybe you can define a unit-tests like this: https://github.com/seladb/PcapPlusPlus/blob/bf438ab7a08a5aac995557f5579a4459adf7a990/Tests/Pcap%2B%2BTest/main.cpp#L1473

Set the configuration to "inflow" and check that all the packets captured are indeed "inflow"

Reply:
Yes, that's good. But the second step was the key to my confusion. How to check that all the packets captured are "inflow"? I only know that I can compare the results with `tcpdump`, but I don't know how to do this in a test function.

Reply:
Yes, the enum definition looks ok. Let's just make sure to clearly mention in the documentation of the enum and `DeviceConfiguration` that this feature is only supported in newer libpcap versions.

Regarding the unit-tests: is it possible to identify inflow packets by MAC address (meaning check that all captured packets have the NIC's MAC address in the dest MAC field)?

Reply:
Thank you for your advice! 
I read the source file of `libpcap`. `pcap_setdirection` was supported since `libpcap-0.9.1`.
I got all the source code from this [website](https://www.tcpdump.org/release/), And `pcap_setdirection` is defined in the file `pcap.c`.

I will complete these modifications as soon as possible.

Reply:
Regarding the unit-tests, I found we could't identify inflow packets by MAC address, because some incoming packets captured by the NIC have different MAC address from the NIC's MAC address, though the program works in normal mode rather than promiscuous mode.

Reply:
Got it. So maybe you can just add a simple unit-test that sets "inflow" in `DeviceConfiguration` and starts capturing packets. It will not test the actual feature but at least it'll check that packet capture is still working.

By the way, I don't see the additional changes on the PR. Please make sure to commit them to the same branch you created the PR.

Also, could you please make these changes on a fork from [dev](https://github.com/seladb/PcapPlusPlus/tree/dev) branch instead of [master](https://github.com/seladb/PcapPlusPlus)?

Reply:
Ok.  I  will do  it  as  soon  as possible.